### PR TITLE
ci: check for unfinished test runs

### PR DESCRIPTION
**Pull request by lenaschoenburg**
Refactors our existing action to run always and adds a new analysis step to find tests which did not finish.

Idea for the script from @deepthidevaki

### CODE REVIEW

**Comment by oleschoenburg** [comment]
Example run that shows the output: https://github.com/camunda/zeebe/actions/runs/7355918829

File: .github/actions/analyze-test-runs/action.yml [hunk 1]
```diff
@@ -39,3 +41,24 @@ runs:
       fi
       echo "No test execution found in build output, thus nothing to check."
       exit 0
+  - name: Unfinished tests
+    if: always()
```
Comment (deepthidevaki) [inline .github/actions/analyze-test-runs/action.yml:45]: :question: Does it needs to be always? Only if test timedout or cancelled, right? But no harm in executing it always though.

**Review by deepthidevaki — APPROVED** [review]
:heart_eyes: Nice.

**Comment by oleschoenburg** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors merge

**Comment by oleschoenburg** [comment]
bors cancel

**Comment by oleschoenburg** [comment]
bors merge

**Pull request merged**